[CI] ClientYamlTestSuiteIT test {yaml=indices.simulate_index_template/10_basic/Simulate index template with data stream with mapping and setting overrides} failing

**Build Scans:**
- [elasticsearch-periodic #9965 / release-tests](https://gradle-enterprise.elastic.co/s/hhmx52rbuldzq)
- [elasticsearch-pull-request #85213 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/el4tjmk6hqjjg)
- [elasticsearch-periodic #9958 / release-tests](https://gradle-enterprise.elastic.co/s/hicnwa3v4634y)
- [elasticsearch-periodic #9951 / release-tests](https://gradle-enterprise.elastic.co/s/v3yfown7whmp2)

**Reproduction Line:**
```
./gradlew ":rest-api-spec:yamlRestTest" --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT" -Dtests.method="test {yaml=indices.simulate_index_template/10_basic/Simulate index template with data stream with mapping and setting overrides}" -Dtests.seed=1DE7951F3BF6C6AA -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=id -Dtests.timezone=Africa/Cairo -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.test.rest.ClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Byaml%3Dindices.simulate_index_template%2F10_basic%2FSimulate%20index%20template%20with%20data%20stream%20with%20mapping%20and%20setting%20overrides%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [indices.simulate_index_template/10_basic:279]: expected [2xx] status code but api [indices.put_data_stream_mappings] returned [400 Bad Request] [{"error":"no handler found for uri [/_data_stream/test/_mappings?error_trace=true] and method [PUT]"}]
```

**Issue Reasons:**
- [main] 3 consecutive failures in step release-tests
- [main] 4 failures in test test {yaml=indices.simulate_index_template/10_basic/Simulate index template with data stream with mapping and setting overrides} (3.7% fail rate in 107 executions)
- [main] 3 failures in step release-tests (100.0% fail rate in 3 executions)
- [main] 3 failures in pipeline elasticsearch-periodic (100.0% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -468,9 +468,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.TextEmbeddingQueryIT
   method: testHybridSearch
   issue: https://github.com/elastic/elasticsearch/issues/132703
-- class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
-  method: test {yaml=indices.simulate_index_template/10_basic/Simulate index template with data stream with mapping and setting overrides}
-  issue: https://github.com/elastic/elasticsearch/issues/132702
 - class: org.elasticsearch.xpack.ml.integration.RevertModelSnapshotIT
   method: testRevertModelSnapshot
   issue: https://github.com/elastic/elasticsearch/issues/132733

